assignments/array-hofs/code.js

```diff
@@ -0,0 +1,59 @@
+//EVENS
+
+const evens = (array) => {
+  return array.filter((n) => n % 2 === 0)
+}
+
+
+//odds
+const odds = (array) => {
+  return array.filter((n) => n & 2 !== 0)
+}
+
+
+//big
+const big = (array) => {
+  return array.filter((n) => n > 100)
+}
+
+//names
+const names = (array) => {
+  return array.map((n) => n.name)
+}
+
+
+//grades
+const grades = (array) => {
+  return array.map((n) => n.grade)
+}
+
+
+//pairs
+const pairs = (array) => {
+  return array.flatMap((n) => n = [[n, n]])
```

Comment:
This works but it would be simpler to use `map` and remove one level of arrays from the value your function returns:

```suggestion
  return array.map((n) => n = [n, n])
```